packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts

```diff
@@ -16,18 +16,18 @@ export function isValidCollectionPathImplementation(
 
   parts.forEach(part => {
     if (passing === false) return
-    if (!part.startsWith(`{`)) return
+    if (!part.includes(`{`) && !part.includes(`}`)) return
```

Comment:
The check in this file was not always correct (it sometimes returned true for invalid syntaxes; I checked that with tests now).

Most important thing here is that `model` is searched for differently. 'Cause it's everything after the first dot. Before it was one string after the last dot.

So with a part of `/{Model.foo.bar.baz}` it'll now be `foo.bar.baz`. Then when it will be asserted with the correct `field` it'll be invalid as `[a-zA-Z_][\w_()]` doesn't allow `.`